tests/Unit/ParallelAlgorithms/ApparentHorizonFinder/Test_ComputeCoords.cpp

```diff
@@ -246,10 +247,252 @@ void test_compute_points() {
     CHECK(current_iteration.compute_coords_retries == 1);
   }
 }
+
+void test_compute_points_different_resolutions() {
+  // As in test_compute_points(), only test grid frame because the only
+  // difference is a different code path is taken within block logical
+  // coordinates which we aren't testing here.
+  using Fr = ::Frame::Grid;
+
+  const size_t l_max = 4;
+  const size_t higher_l_max = 6;
+  ah::Storage::Iteration<Fr> current_iteration{};
+
+  const domain::creators::Sphere sphere{
+      1.0,          3.0,  domain::creators::Sphere::Excision{nullptr},
+      0_st,         2_st, true,
+      std::nullopt, {2.0}};
+  const Domain<3> domain = sphere.create_domain();
+  const domain::FunctionsOfTimeMap functions_of_time =
+      sphere.functions_of_time();
+
+  const size_t max_compute_coords_retries = 3;
+  const LinkedMessageId<double> time{.id = 4.0, .previous = {3.0}};
+  ylm::Strahlkorper<Fr> initial_guess{l_max, 2.5, std::array{0.0, 0.0, 0.0}};
+  ylm::Strahlkorper<Fr> previous_iteration_surface =
+      current_iteration.strahlkorper;
+  std::deque<ah::Storage::PreviousSurface<Fr>> previous_surfaces{};
+  const FastFlow fast_flow{
+      FastFlow::FlowType::Fast, 1.0, 0.5, 1.e-12, 1.e-2, 1.2, 5, 100};
+
+  // Test case 1: current_resolution_l is set and different from initial_guess
+  {
+    current_iteration.block_coord_holders.reset();
+    current_iteration.strahlkorper = ylm::Strahlkorper<Fr>{};
+    current_iteration.compute_coords_retries = 0;
+    initial_guess =
+        ylm::Strahlkorper<Fr>{l_max, 2.5, std::array{0.0, 0.0, 0.0}};
+
+    const bool coords_set_successfully = set_current_iteration_coords(
+        make_not_null(&current_iteration), time, fast_flow, initial_guess,
+        previous_iteration_surface, previous_surfaces,
+        max_compute_coords_retries, domain, functions_of_time, higher_l_max,
+        false);
+
+    CHECK(coords_set_successfully);
```

Comment:
Should this be `REQUIRE`? I assume the rest of the test is nonsensical if this check fails? Some with `CHECK`s on 300, 301?